src/pocketmine/utils/MainLogger.php

```diff
@@ -209,20 +222,20 @@ public function logException(\Throwable $e, $trace = null){
 		$errstr = preg_replace('/\s+/', ' ', trim($errstr));
 		$errfile = Utils::cleanPath($errfile);
 
-		$message = get_class($e) . ": \"$errstr\" ($errno) in \"$errfile\" at line $errline";
+		$message = $induced ? "Caused by: " : "";
+		$message .= get_class($e) . ": \"$errstr\" ($errno) in \"$errfile\" at line $errline";
+		yield [$type, $message];
```

Comment:
Don't think a type is necessary here. Previous exceptions can be logged in debug the same as the rest. Having it as its own error will make it confusing to read, since it's hard to tell if the debug blocks are part of the same log error.

Additionally, consider referring to this as "previous" instead of "caused by" to reduce ambiguity when there are multiple chained exceptions.

Reply:
Also, it's possible to yield the message type as key if you want to keep it.

Reply:
About the key: I use iterator_to_array() to prevent running the message generation in the synchronized() block, so I'm not using the key here.

Reply:
Since iterator_to_array() has been removed, using the type as key is more convenient (but it's the same thing anyway!)